exporter/splunkhecexporter/logdata_to_splunk.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecexporter
+
+import (
+	"math"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+const (
+	sourceLabel     = "service.name"
+	sourcetypeLabel = "source.type"
+)
+
+func logDataToSplunk(logger *zap.Logger, ld pdata.Logs, config *Config) ([]*splunkEvent, int) {
+	numDroppedLogs := 0
+	splunkEvents := make([]*splunkEvent, 0)
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		if rl.IsNil() {
+			continue
+		}
+
+		ills := rl.InstrumentationLibraryLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ils := ills.At(j)
+			if ils.IsNil() {
+				continue
+			}
+
+			logs := ils.Logs()
+			for j := 0; j < logs.Len(); j++ {
+				lr := logs.At(j)
+				if lr.IsNil() {
+					numDroppedLogs++
```

Comment:
Should it really be considered "dropped" if it never existed?

Reply:
Good point, will move it down to check if body is nil.